azerothcore/azerothcore-wotlk
# Feat(Core/Gossips): Override BoxMoney from DB

## Changes Proposed:
-  Add a new param to AddGossipItemFor (DB data), to allow to override BoxMoney data. This would be useful for trainers, which the box money depends on the player level. Currently, the only way to change it is from DB and is a const value.

## Issues Addressed:

None related. 

## SOURCE:


## Tests Performed:

- Tested in-game with https://github.com/azerothcore/azerothcore-wotlk/pull/10415

## How to Test the Changes:


1. Having https://github.com/azerothcore/azerothcore-wotlk/pull/10415, you can go to any npc. i.e.: .go c id 7870
2. Level Leatherworking to 225 and be level 40.
3. .learn 10660 (tribal leatherworking)
4. Talk to the npc and unlearn tribal leatherworking. (At level 40, you should see 50g on the pop-up box)
5. .learn 10660 again
6. .char level 70
7. Now, talk to the npc again to unlearn it. You should see 100g on the pop-up box.
